### PR TITLE
ignore files in vcs (eg, .git) direcotry

### LINKED ISSUE #7084: az acr build should exclude .git folder
**Describe the bug**
If the source code folder has the vcs folders eg `.git` or `.bzr` , az acr build should exclude the files from the folders in the uploaded tar file.

**To Reproduce**
Source code folder structure:
.git 
 | -- file1
Dockerfile

az acr build -t <registry> <source_code_folder>

**Expected behavior**
The uploaded tar file should exlude `.git` folder. The tar file should only have `Dockerfile`.

**Additional context**
It's a regression from #6950 

Related issue: https://github.com/Azure/acr/issues/122

If users manually add .git in .dockerignore file. The .git folder will be excluded.

### PR DESCRIPTION
**Pull request by northtyphoon**
---

Fix #7084 

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by northtyphoon** [comment]
@djyou

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py [hunk 1]
```diff
@@ -401,11 +401,13 @@ def _pack_source_code(source_location, tar_file_path, docker_file_path):
     def _ignore_check(tarinfo, parent_ignored, parent_matching_rule_index):
         # ignore common vcs dir or file
         if tarinfo.name in common_vcs_ignore_list:
-            logger.debug(".dockerignore: ignore vcs file '%s'", tarinfo.name)
+            logger.warning("Excluding '%s' based on default ignore rules", tarinfo.name)
```
Comment (djyou) [inline src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py]: Is this logging going to be overwhelming?

Comment (northtyphoon) [reply src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py]: It only logs top folder and file name.

**Comment by ehotinger** [comment]
@yugangw-msft  / @tjprescott  / @troydai  PTAL and merge if it looks good. This is an important fix for customers.

**Pull request merged**